### ISSUE TITLE
Weird Behavior in Classic

### ISSUE THREAD

**Issue by Gogo1951** [arkayenro/arkinventory#1485]
I pick up 2 items from the mail, and they don't show in my bags.

![image](https://user-images.githubusercontent.com/55365231/127800115-74fff82d-0484-4c61-89eb-7252b8714fc5.png)

I load the game, and one of my mounts doesn't appear in the mounts spot. 

![image](https://user-images.githubusercontent.com/55365231/127800156-237e1d56-8722-4081-ad3a-0d744774d726.png)

In both cases, things get right after a reload UI.

Just started happening a few days ago.

**Comment by arkayenro** [comment]
what version do you have, and if you go into edit mode do the white squares have data when you click on them?  and does the debug info for the mount show what category it is in?

**Comment by Gogo1951** [comment]
Mount

![image](https://user-images.githubusercontent.com/55365231/128124184-b5891f43-6db6-461b-8b6a-329fdaf2d62f.png)

White Spaces (Should be Green Items to DE)

![image](https://user-images.githubusercontent.com/55365231/128124228-4f61eb30-af2b-4064-9f9b-d52676f9bd33.png)

**Comment by Gogo1951** [comment]
Green Items Should have Been:

![image](https://user-images.githubusercontent.com/55365231/128124278-15fdff2a-c871-4191-aa95-f9a52b65743e.png)

**Comment by Gogo1951** [comment]
Version

![image](https://user-images.githubusercontent.com/55365231/128124343-de20b331-15ad-4d02-ba18-842f90a0b9ea.png)

**Comment by Gogo1951** [comment]
Is there another debug too I can pull up?

**Comment by Gogo1951** [comment]
Oops. Here's an item:

![image](https://user-images.githubusercontent.com/55365231/128124413-1c3b8aa2-54c9-48bc-a5e4-4231c8e1506d.png)

And here's the debug on the mount:

![image](https://user-images.githubusercontent.com/55365231/128124441-7b4999c9-6dbc-49fe-99bd-317b2d4be91e.png)

**Comment by Gogo1951** [comment]
When I move the items by clicking on the white boxes, they show up in bags. 

![image](https://user-images.githubusercontent.com/55365231/128124605-2fbbc653-466c-4a14-841e-22c7da919fc8.png)

**Comment by Gogo1951** [comment]
When I move the mount... it clones the icon. Only once though.

![image](https://user-images.githubusercontent.com/55365231/128124654-7c623a9d-eb69-41e0-9083-696379e399e8.png)

**Comment by Gogo1951** [comment]
![image](https://user-images.githubusercontent.com/55365231/128387953-5cb58c11-7ff3-49db-b128-6fd631a58fd1.png)

Seems to be happening with random stuff. Trying to figure out if it's by "real" bag location or something. I don't see a pattern.

**Comment by Gogo1951** [comment]
Oh, so it's not reading "soulbound" correctly. Here's the same hearthstone after a reloadui.

![image](https://user-images.githubusercontent.com/55365231/128388148-18799bcb-8a82-4511-8d3e-491afab3d43f.png)

**Comment by arkayenro** [comment]
the "(temporary) retrieving item data" as the name means one (or more) of the items havent been cached yet but once it has the data just clicking on refresh should update it

i cant seems to get it to happen on my bcc character though, so its making it hard to work out

**Comment by Gogo1951** [comment]
Are there logs I can provide, or any sort of debug interface? Would a screenshare help?

**Comment by arkayenro** [comment]
not really.  its obviously happening but its not consistent.

with the game shut down can you rename the _classic_\Cache folder to Cache.old (or delete it).  it will rebuild everything and that might help

**Comment by Gogo1951** [comment]
Still happening after deleting cache folders.

**Comment by Feirlane** [comment]
Same thing is happening to me today after `v3.09.54 [The Burning Crusade]` made it to the stable release. A consistent way of reproducing it seems to be moving an item in the "Default" section of Arkinventory to a slot backed up by the default bag. Doing that makes the object invisible, but dragging the emtpy slot to a slot backed by a different bag brings it back. See [this gif](https://i.imgur.com/cv7wQjm.mp4) for an example.

**Comment by arkayenro** [comment]
that works fine for me.  can you export the profile you are using and send it to me?

**Comment by Feirlane** [comment]
Profile for the video above: https://pastebin.com/EPrNLtTx

**Comment by Feirlane** [comment]
Some more information trying to reproduce the disappearing items: if I `/reload` or log out and back in again, the issue is gone. However if I fully close the game and log back in the issure re-appears.

**Comment by arkayenro** [comment]
could you try [r1145-alpha](https://www.curseforge.com/wow/addons/ark-inventory/files/3425036) and let me know if its fixed or not

**Comment by Feirlane** [comment]
The same thing happens with that version :( , and as other have mentioned the count of stacks also doesn't change when clicking on them, for example after drinking some water. Any debug version/logs I can send you to analyze this behaviour?

**Comment by arkayenro** [comment]
try sending me the profile again, its gone from pastebin

**Comment by arkayenro** [comment]
are you using the normal version or the nolib version?

**Comment by Gogo1951** [comment]
Unfortunately still happening for me as well.

![image](https://user-images.githubusercontent.com/55365231/129447138-e19be501-93bf-472c-b8b9-80bf4d97f378.png)

**Comment by Feirlane** [comment]
Let's see if [this gist with the profile](https://gist.github.com/Feirlane/9d952f454589f81c0f0ff2a1f1cf7b89) last for longer than the pastebin. 

I'm using the default one downloaded by WowUp, which includes the `Externals` folder, so I assume it's the regular one.

**Comment by Gogo1951** [comment]
```

^1^T^Sprofile^T^Sname^SDefault^Sguid^S4731-01DF9AC9-20210717035523-7334-900D-ECC5^Slocation^T^N1^T^Sanchor^T^Sb^F4600483352150016^f-43^Sl^F5750751292817408^f-42^St^F4644400466493440^f-42^Sr^N2099.5693359375^t^Scontainer^T^Sheight^N355^Swidth^N792^t^t^N2^T^Sanchor^T^Sb^F7545095213547520^f-43^Sl^F4589607421214720^f-43^St^F5004000361447426^f-42^Sr^F5637319595917312^f-42^t^Scontainer^T^Sheight^N159^Swidth^N760^t^t^N3^T^Sanchor^T^Sb^F6023717939314688^f-44^St^F4641736513028096^f-42^Sl^F8957709420331012^f-43^Sr^F8032475754266624^f-42^t^Scontainer^T^Sheight^N535^Swidth^N808^t^t^N10^T^Sanchor^T^Sb^F5062152071151618^f-42^Sl^N1875.5556640625^Sr^N2275.5556640625^St^N1280^t^Scontainer^T^Sheight^N8^t^t^t^Sused^SY^t^Scatset^T^N1000^T^Sname^SDefault^Sguid^S4731-01DF9AC9-20210717035523-0A1D-D269-E60C^Sused^SY^t^t^Sdesign^T^N1000^T^Sslot^T^Spad^N8^Sborder^T^Soffset^N5^Sstyle^SArkInventory~`Tooltip~`3^Ssize^N16^t^Sunusable^T^Stint^B^t^t^Sguid^S4731-01DF9AC9-20210717035523-16A6-1206-00D8^Sused^SY^Scategory^T^S1!201^N3^S1!111^N2^S1!411^N4^S1!209^N3^S1!107^N2^S1!108^N2^S1!203^N3^S1!403^N4^S1!404^N3^S1!104^N2^S1!512^N2^S1!102^N2^S1!420^N3^S1!426^N3^S1!424^N3^S1!409^N8^S1!501^N2^S1!208^N3^S1!415^N2^S1!432^N3^S1!101^N2^S1!418^N3^S1!110^N2^S1!417^N3^S1!112^N2^S1!437^N3^S1!310^N10^S1!416^N5^S1!503^N2^S1!206^N3^S1!202^N3^S1!423^N6^S1!436^N3^S1!513^N2^S1!113^N2^S1!438^N4^S1!506^N2^S1!502^N2^S1!106^N2^S1!103^N2^S1!902^N3^S1!205^N3^S1!504^N2^S1!505^N2^S1!428^N3^S1!412^N2^S1!402^N9^S1!427^N2^S1!430^N3^S1!204^N3^S1!105^N2^S1!421^N10^S1!207^N3^S1!431^N3^S1!433^N3^S1!435^N3^S1!419^N3^t^Sname^SDefault^Sbar^T^Sdata^T^N1^T^Ssort^T^Smethod^N1001^t^t^N2^T^Ssort^T^Smethod^N1001^t^t^N3^T^Ssort^T^Smethod^N1001^t^t^N4^T^Ssort^T^Smethod^N1001^t^t^N5^T^Ssort^T^Smethod^N1001^t^t^N6^T^Ssort^T^Smethod^N1001^t^t^N8^T^Ssort^T^Smethod^N1001^t^t^N9^T^Ssort^T^Smethod^N1002^t^t^N10^T^Ssort^T^Smethod^N1001^t^t^t^t^t^t^Scat^T^t^Ssort^T^N1001^T^Sname^SGogo^Sguid^S4731-01DF9AC9-20210717035523-6E28-B76C-A625^Sorder^T^N1^T^Skey^Sslottype^t^N2^T^Skey^Sexpansion^t^N3^T^Skey^Sbagid^t^N4^T^Skey^Squality^Sdescending^B^Sactive^B^t^N5^T^Skey^Sitemstatlevel^Sdescending^B^Sactive^B^t^N6^T^Skey^Sid^t^N7^T^Skey^Sitemuselevel^t^N8^T^Skey^Svendorprice^t^N9^T^Skey^Sitemage^t^N10^T^Skey^Sitemtype^t^N11^T^Skey^Sslotid^t^N12^T^Skey^Scategory^t^N13^T^Skey^Slocation^t^N14^T^Skey^Sname^Sactive^B^t^N15^T^Skey^Scount^Sdescending^B^Sactive^B^t^t^Sused^SY^t^N1002^T^Sname^SPrice^Sguid^S4731-01DF9AC9-20210717035523-7E43-E9A7-D34C^Sorder^T^N1^T^Skey^Sslottype^t^N2^T^Skey^Sexpansion^t^N3^T^Skey^Sbagid^t^N4^T^Skey^Sitemstatlevel^t^N5^T^Skey^Sid^t^N6^T^Skey^Sitemuselevel^t^N7^T^Skey^Svendorprice^Sdescending^B^Sactive^B^t^N8^T^Skey^Sitemage^t^N9^T^Skey^Sitemtype^t^N10^T^Skey^Sslotid^t^N11^T^Skey^Scategory^t^N12^T^Skey^Slocation^t^N13^T^Skey^Sname^t^N14^T^Skey^Scount^t^N15^T^Skey^Squality^t^t^Sused^SY^t^t^t^^

```

**Comment by arkayenro** [comment]
try alpha 3 / [r1146-alpha](https://www.curseforge.com/wow/addons/ark-inventory/files/3425220) and let me know if its fixed or not.

**Comment by arkayenro** [comment]
note, if you shift the sorting from instant to on open youll probably get rid of most of the issue

**Comment by Feirlane** [comment]
Both issues still happening with alpha 3, moving items to the default bag makes them dissapear (but can be dragged back into existance from the empty slot), and drinking a water does not update the stack count. No messages in the chat window with the debug build and debug enabled. Same behaviour with either instant sorting or "on window open"

**Comment by AmosPolanski** [comment]
I have been having this same issue @Feirlane has described.  This is a weird bug.  Out of 8 toons on the same account 6 have issues with the default bag, only the default bag.  Items disappear when moved into the default bag, gray boxes where loot should be in the default bag and icons still showing up after moving items out of the default bag.

I tried the alpha build and still have the same issue with the 6 toons.  I tried a clean install, deleted the addon saved variable files and then reinstalled. Still have the same issue.  @arkayenro if you need any files to try and help trouble shoot this issues please let me know and I will be happy to provide them.

**Comment by Feirlane** [comment]
Same behaviour with Alpha 5, no messages showing up in the chat window when moving objects or drinking consumes.

**Comment by arkayenro** [comment]
make sure youre using alpha 7
click on switch character, mouse over the current character, select erase backpack data. see if that helps.

**Comment by AmosPolanski** [comment]
I am still getting ghost icons and invisible items in the default backpack after erasing all backpack data with alpha 7.

![image](https://user-images.githubusercontent.com/41087510/129582698-116eeee2-2281-4ad9-aa3d-36fbdb896201.png)

**Comment by Gogo1951** [comment]
I cleared out all my data across all my toons, and I haven't been able to get the issue to happen since. The "fix" just means I have to re-log in to reset all my data, seems easy enough. Haven't had it happen again since resetting and re-caching all my data. Thanks!

![image](https://user-images.githubusercontent.com/55365231/129636758-978353b7-1c7e-42da-89b0-5a776f1cd7bd.png)

**Comment by Gogo1951** [comment]
Spoke too soon!

Worked on all my toons except my real hunter, not my bank hunter. =P

![image](https://user-images.githubusercontent.com/55365231/129636969-86ef6d44-fe0c-4426-8d22-8d5f0097baf9.png)

![image](https://user-images.githubusercontent.com/55365231/129636981-93b241a9-3591-4cf6-bab8-124a757964b3.png)

![image](https://user-images.githubusercontent.com/55365231/129637004-08baec5b-7ecb-4e8d-8c3b-d207165d4a3f.png)

**Comment by Gogo1951** [comment]
Just did a totally fresh install, cleared all saved variables, and re-installed. Copied over my old profile settings via in-app export feature... and still having the issue as soon as I log out and log back in again. =P

**Comment by arkayenro** [comment]
go into your profile and make sure that backpack data is being saved.  if you reset everything then it should be

if i turn that off i can duplicate the issue.  the actual issue is the backpack stops getting scanned but i cant figure out why thats happening

**Comment by arkayenro** [comment]
should be fixed in [alpha 8](https://www.curseforge.com/wow/addons/ark-inventory/files/3428782)

**Comment by Feirlane** [comment]
Both issues (moving an item to the default bag, and the stack count not updating when using consumes) seem to be fixed for me with Alpha 8. Thanks a lot!

**Comment by AmosPolanski** [comment]
Alpha 8 seems to have resolved all the issues for me as well.  Thanks!

**Comment by Gogo1951** [comment]
![image](https://user-images.githubusercontent.com/55365231/129807831-69e8280e-6719-4b60-a692-89a516176b90.png)

I'm still seeing the issue. Deleted all my saved variables, did a fresh install, and restored old profile from import... only see it on a few toons, hunter being one.

**Comment by arkayenro** [comment]
check [alpha 10](https://www.curseforge.com/wow/addons/ark-inventory/files/3430371), that should fix the last of the issues i know about

**Comment by Gogo1951** [comment]
I'm on 11. =(

![image](https://user-images.githubusercontent.com/55365231/130337329-4f67b7a0-d5d6-4d3e-b86a-9f5df913d61a.png)

**Comment by arkayenro** [comment]
theres no grey slots in the image, what issue am i looking for?

**Comment by Gogo1951** [comment]
Oh, the items in the lower right corner should be in other bags. Instead they are incorrectly being put in the lower right corner of my general bag. Everything should be anchored top left as well. It's a bit odd.

When I enter the game.

![image](https://user-images.githubusercontent.com/55365231/130338069-2dd572bc-d54d-40df-8626-c7d7d982b684.png)


Then when I /reload it looks like this -- the way it should look.

![image](https://user-images.githubusercontent.com/55365231/130338079-9b75150d-1b72-49d7-8bda-c3528a11b688.png)

**Comment by Gogo1951** [comment]
It's not the same issues as the ghost boxes. It's just that somehow the items aren't getting tagged correctly as the right category so they can't be placed. I think you had said it was a similar issue in the past. Let me know if it's a separate issue. Thanks!

**Comment by arkayenro** [comment]
if you click on refresh do they end up in the correct locations?
if you go into edit mode do they move to the correct locations?
what is their assigned and default category values (when in edit mode)?

**Comment by Gogo1951** [comment]
On Load.

![image](https://user-images.githubusercontent.com/55365231/130340098-86b7823d-1bc3-4538-b7b1-fd423e55733b.png)

![image](https://user-images.githubusercontent.com/55365231/130340102-2e36ea84-e518-4cd9-87e9-55144022cad3.png)

![image](https://user-images.githubusercontent.com/55365231/130340103-494a93f4-1c4e-4264-aa1c-bddbf1bf4ed2.png)

![image](https://user-images.githubusercontent.com/55365231/130340107-3b1d4e83-0c52-48d0-86ee-bbbcce71cd23.png)

**Comment by Gogo1951** [comment]
Gloves after reload:

![image](https://user-images.githubusercontent.com/55365231/130340136-6ed6bb2d-343b-4e44-aaf7-a8630ae3f071.png)

**Comment by arkayenro** [comment]
try moving them to another slot and see if they update correctly

**Comment by Gogo1951** [comment]
If I move them, I have to open and close the bag but they show up in the right spot after. Or if I /reload they show up in the right spot.

**Comment by arkayenro** [comment]
try [alpha 12](https://www.curseforge.com/wow/addons/ark-inventory/files/3434087)

**Comment by Gogo1951** [comment]
Seems to be fixed! (=
